tests/test_checks.py

```diff
@@ -1366,4 +1366,25 @@ def test_fail_list_of_constraints(self):
                 .run()
 
         df = VerificationResult.checkResultsAsDataFrame(self.spark, result)
-        self.assertEqual(df.select("constraint_status").collect(), [Row(constraint_status="Success"), Row(constraint_status="Success")])
\ No newline at end of file
+        self.assertEqual(df.select("constraint_status").collect(), [Row(constraint_status="Success"), Row(constraint_status="Success")])
+
+    def test_hash_code(self):
+        """
+        Lack of Exception is passing.  Previously this test would fail with:
+        AttributeError: 'ScalaFunction1' object has no attribute 'hashCode'
+        """
+        vrb = VerificationSuite(self.spark) \
+            .onData(self.df)
+        check = Check(self.spark, CheckLevel.Error, "Enough checks to trigger a hashCode not an attribute of ScalaFunction1")
+        check.isComplete('b')
+        vrb.addCheck(check)
+        check.containsEmail('email')
+        vrb.addCheck(check)
+        check.isGreaterThanOrEqualTo("d", "b")
+        vrb.addCheck(check)
+        check.isLessThanOrEqualTo("b", "d")
+        vrb.addCheck(check)
+        check.hasDataType("d", ConstrainableDataTypes.String, lambda x: x >= 1)
+        vrb.addCheck(check)
```

Comment:
Why do we need to add and verify one by one?

Reply:
Because that is the use case for triggering the exception.  If I change the test like so, it does not use `hashCode`.

```
vrb = VerificationSuite(self.spark) \
            .onData(self.df)
        check = Check(self.spark, CheckLevel.Error, "Enough checks to trigger a hashCode not an attribute of ScalaFunction1")
        check.addConstraints([
            check.isComplete('b'),
            check.containsEmail('email'),
            check.isGreaterThanOrEqualTo("d", "b"),
            check.isLessThanOrEqualTo("b", "d"),
            check.hasDataType("d", ConstrainableDataTypes.String, lambda x: x >= 1)])

        result = vrb.addCheck(check).run()
```

Reply:
The issue write up has more info: https://github.com/awslabs/python-deequ/issues/91

Reply:
I can't replicate the error in CI but submitted an attempt to fix it.